### DISCUSSION TITLE
Is there a way to check if an inline partial is defined?

### DISCUSSION THREAD

**Discussion by peter-affenzeller** [handlebars-lang/handlebars.js#1783] [Q&A]
in the [inline partials example](https://handlebarsjs.com/examples/partials/inline-blocks.html), the inline partials of nav and content are each wrapped inside a div.
if one of those partials is not defined, the wrapping div will still be created, which in my case leads to issues since I am using a grid container, which is creating a new row for the div and hence also gaps before/after the new row, even though there is nothing in it.

is there some way to check if the inline partial actually exists?
something like:
```
{{#if (hasInlinePartial "nav"}}
<div class="nav">{{> nav}}</div>
{{/if}}
```

I couldn't find anything in the options passed to helper functions...

**Comment by illepic** [comment]
Checking if you found a solution to this?

**Reply by peaffenzeller** [reply]
unfortunately not, no, but wasn't too bad in my case, we ended up always having all the inline partials, else I would have had some useless wrapping divs introducing equally useless spacings, but could have probably worked my way around that with a couple more css rules...
what would work (even though not an ideal solution...) is to just add more props to the partial ("parent" partial, not the inline partials), like ```hasContent``` or ```hasCTA```, depending on the inline partials, and check for those.

**Reply by rpaterson** [reply]
Just implemented the `hasInlinePartial` hack suggested above which worked for me. Pity there's not a more elegant way to do this.

**Reply by Aleksandern** [reply]
@rpaterson 
Could you share your hack?

**Reply by rpaterson** [reply]
In addition to the inline partial I pass in a boolean flag.  Like this:
```handlebars
{{#> myPartial hasMyInlinePartial=true}}
    <other-html-1>
    {{#* inline "myInlinePartial" }}
        <some-html>
    {{/ inline}}
    <other-html-2>
{{/myPartial}}
```
or
```handlebars
{{#> myPartial}}
    <other-html-1>
    <other-html-2>
{{/myPartial}}
```

Then in `myPartial.hbs`:
```handlebars
{{#if hasMyInlinePartial}}
    <div>{{> myInlinePartial}}</div>
{{/if}}
```

**Reply by Aleksandern** [reply]
Well, I came to the same decision. Thanks.

**Comment by jaket-hub** [comment]
Not used handlebars out of the box myself - used via foundation zurb panini

I was testing the"Inline partial" and used with"Partial Blocks Fallover" to cater for this but was looking for an alternative inbuilt solution.
Was wondering if there was a way to detect if a partial exists - apparently not.

Any how the "Partial Blocks Fallover" can be an alternative technique which could be used.
Make the fallover content empty.
https://handlebarsjs.com/examples/partials/failover.html

The inline partial would need to cater for any wrapper markup.

Something like this - https://handlebarsjs.com/examples/partials/inline-blocks.html modified
```handlebars
{{#> layout}}
  {{#*inline "nav"}}<div class="nav">My Nav</div>{{/inline}}
  {{#*inline "content"}}My Content{{/inline}}
{{/layout}}
```
```handlebars
{{#> nav}}No nav specified{{/nav}}
<div class="content">{{> content}}</div>
```
